src/transforms/code-scanning-alert.ts

```diff
@@ -1,35 +1,80 @@
 import issueKeyParser from "jira-issue-key-parser";
-import { Context } from "probot/lib/context";
-import { JiraRemoteLinkData } from "../interfaces/jira";
+import {Context} from "probot/lib/context";
+import {JiraRemoteLinkData} from "../interfaces/jira";
 
-function getLastUpdated({ status, started_at, completed_at }) {
-	if (status === 'completed') {
-		return completed_at;
-	} else {
-		return started_at;
+// TODO ARC-618
+const getPullRequestTitle = (id: number): string => {
+	return `GH-${id}`;
+}
+
+const getEntityTitle = (ref: string): string => {
+	// ref can either be a branch reference or a PR reference
+	const components = ref.split("/")
+	switch (components[1]) {
+		case "heads": // branch
+			// The branch name may contain forward slashes! Rejoin them
+			return components.slice(2).join("/");
+		case "pull": // pull request
+			return getPullRequestTitle(parseInt(components[2]));
+		default:
+			// log something here
+			return "";
+	}
+}
+
+const transformStatusToAppearance = (status: string): string => {
+	switch (status) {
+		case "open":
+			return "removed"; // red
+		case "fixed":
+			return "success"; // green
+		case "dismissed":
+			return "moved"; // yellow
+		default:
+			return "default";
 	}
 }
 
 export default (context: Context): JiraRemoteLinkData => {
-	const { code_scanning_alert } = context.payload;
-	const issueKeys = issueKeyParser().parse(`${code_scanning_alert.instances.ref}`);
+	const { action, alert, ref, repository} = context.payload;
+
+	// Grab branch names or PR titles
+	const entityTitles = [];
+	if (action === "closed_by_user" || action === "reopened_by_user") {
+		// These are manual operations done by users and are not associated to a specific Issue.
+		// The webhook contains ALL instances of this alert, so we need to grab the ref from each instance.
+		entityTitles.push(...alert.instances.map((instance) => getEntityTitle(instance.ref)));
+	} else {
+		// The action is associated with a single branch/PR
+		entityTitles.push(getEntityTitle(ref));
+	}
+
+	const parser = issueKeyParser();
+	// Flatten and get all issue keys
+	const issueKeys = [].concat(...entityTitles.map((entityTitle) => parser.parse(entityTitle)));
 
-	if (!issueKeys) {
+	if (issueKeys.length === 0) {
 		return undefined;
 	}
 
 	return {
-		//todo mapping + add issueKeys into associations
-		product: 'GitHub - Code Scanning Alert',
 		remoteLinks: [{
-			schemaVersion: '1.0',
-			id: code_scanning_alert.external_id,
+			schemaVersion: "1.0",
+			id: `${repository.id.toString()}-${alert.number}`,
 			updateSequenceNumber: Date.now(),
-			associations: [],
-			displayName: code_scanning_alert.name,
-			url: code_scanning_alert.html_url,
-			type: 'security',
-			lastUpdated: getLastUpdated(code_scanning_alert)
-		}],
+			displayName: alert.rule.description,
+			description: alert.rule.full_description,
```

Comment:
I was just checking on the UI and it doesn't seem Jira is able to display the whole description. It also has a limit of 255 characters. We should consider displaying just the `alert.rule.description` here and something else as the link. instead.

Reply:
Both displayname and description can be hovered over to get more information:
![image](https://user-images.githubusercontent.com/3264377/132450051-14dda9aa-cb8e-4e3c-8a07-ebacd1ec4355.png)

But you're right, this could go over 255 chars and the request will fail, so I need to probably slice everything after 255.

I'm not against changing the values around but maybe need to ask PM.

Reply:
Oh cool I didn't know we could hover over! But yeah, we'll still need to slice it

Reply:
I've added a slice